### DISCUSSION TITLE
How to stack 2D H&E Images to 3D to see volume

### DISCUSSION THREAD

**Discussion by chokevin8** [marcomusy/vedo#709] [Q&A]
Hi, so I've started to read the documentation for vedo, but I'm still confused on how to start, so I was wondering if anyone could help. I have about 30 .png images of 2-D RGB H&E Images (so they are all 3-d numpy arrays of size (x,y,3)) and for visualization, I'm trying to stack these 2-D images on top of each other in the z-axis so that I have a 3-D visualization of these images. Could someone help me how to perhaps start on this? All of the 2-D RGB H&E Images have the same size and they have already been registered so that simply stacking them in 3-D will result in a visualization. Thank you so much!

**Comment by marcomusy** [comment]
If already have the numpy array then it's easy:
```python
from vedo import *
vol = Volume(array)
vol.print()
show(vol, axes=1)
```

**Reply by chokevin8** [reply]
Using this code, I would only be able to use just one 2-D image in vedo. Maybe I wasn't clear enough in what I wanted. How would I stsack these multiple 2-D images together?

**Reply by marcomusy** [reply]
Hi, the array can be 3D with RGB channels (but you need to average somehow the RGB values or pick only one if you want to visualize it..),
so consider this example:
```python
from vedo import *

array = np.random.rand(10,10,10,3)*255  # your initial data stacked array

array = np.mean(array, axis=3) # average rgb intensities
print(array.shape)

vol = Volume(array)
vol.print()
show(vol, axes=1)
```
![Screenshot from 2022-10-20 20-37-03](https://user-images.githubusercontent.com/32848391/197030853-d86a7005-8c12-4c44-9938-4044b7249931.png)
